src/components/_internal/BaseTabs/BaseTabLabel.tsx

```diff
@@ -74,15 +74,26 @@ const segmentedSizes = {
   [BaseTabSizes.md]: mediumSegmentedSize,
 };
 
+const commonHoverTab = css<BaseLabelProps>`
+  ${({ $isSelected, $color, theme }) =>
+    $color && !$isSelected
+      ? lighten(0.1, getColor($color, { theme }))
+      : getColor('primary.600')}
+`;
+
 const underlineTab = css<BaseLabelProps & { size: Sizes }>`
   ${({ size }) => underlineSizes[size] || underlineSizes[BaseTabSizes.md]};
   padding-bottom: ${getSpace(SpaceSizes.xxs)};
   border-bottom: 2px solid
     ${({ $isSelected, $color }) =>
-      $isSelected ? getColor($color) : getColor('neutral.500')};
+      $isSelected
+        ? getColor('primary.500')
+        : $color
+        ? getColor($color)
+        : getColor('neutral.500')};
```

Comment:
I don't think this is correct. It should work that the `$color` is the selected color, if `$color` is not provided then we use the default primary. It's not in Figma but this will break the IHS report modal where we are using their brand color for this special report

Reply:
oh got it. I thought the custom color was for the normal case. And that's the reason why I've changed the test. Fixing it